plugins/org.eclipse.n4js/src/org/eclipse/n4js/validation/validators/N4JSExpressionValidator.xtend

```diff
@@ -566,14 +566,21 @@ class N4JSExpressionValidator extends AbstractN4JSDeclarativeValidator {
 		if (typeRef instanceof UnknownTypeRef)
 			return; // suppress error message in case of UnknownTypeRef
 
+		val G = newExpression.newRuleEnvironment;
 		// not even a TypeTypeRef?
 		if (! (typeRef instanceof TypeTypeRef)) {
+			if (typeRef.isDynamic) {
+				val constrOfWildcard = TypeUtils.createTypeTypeRef(TypeUtils.createWildcard, true);
+				if (ts.subtypeSucceeded(G, typeRef, constrOfWildcard) || ts.subtypeSucceeded(G, constrOfWildcard, typeRef)) {
+					// don't bother on dynamic types that are either a sub- or supertype of function
```

Comment:
Not `Function` but `constructor{?}` at the end of the comment. And maybe "with" instead of "on"??
```
// don't bother with dynamic types that are either a sub- or supertype of constructor{?}
```